superdesk/io/commands/update_ingest.py

```diff
@@ -104,8 +104,40 @@ def get_provider_rule_set(provider):
 
 
 def get_provider_routing_scheme(provider):
-    if provider.get('routing_scheme'):
-        return superdesk.get_resource_service('routing_schemes').find_one(_id=provider['routing_scheme'], req=None)
+    """Returns the ingests provider's routing scheme configuration.
+
+    If provider has a routing scheme defined (i.e. scheme ID is not None), the
+    scheme is fetched from the database. If not, nothing is returned.
+
+    For all scheme rules that have a reference to a content filter defined,
+    that filter's configuration is fetched from the database as well and
+    embedded into the corresponding scheme rule.
+
+    :param dict provider: ingest provider configuration
+
+    :return: fetched provider's routing scheme configuration (if any)
+    :rtype: dict or None
+    """
+
+    if not provider.get('routing_scheme'):
+        return None
+
+    schemes_service = superdesk.get_resource_service('routing_schemes')
```

Comment:
OK, good, I'll see how independent content filters are from the rest of the code, but at a first glance I don't expect too many problems (I cannot say much about the routing schemes, though, I might skip them since you say that incremental changes are also fine).

I will update the PR on Monday.